sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_path_client.py

```diff
@@ -392,6 +392,156 @@ def get_access_control(self, upn=None,  # type: Optional[bool]
+        :rtype: :class:`~azure.storage.filedatalake.models.AccessControlChanges`
+        """
+        if not acl:
+            raise ValueError("The Access Control List must be set for this operation")
+
+        options = self._set_access_control_recursive_options('modify', acl=acl, **kwargs)
+        return self._set_access_control_internal(options, progress_callback)
+
+    def remove_access_control_recursive(self,
+                                        acl=None,  # type: Optional[str]
+                                        progress_callback=None,
+                                        **kwargs):
+        # type: (...) -> Dict[str, Union[str, datetime]]
+        """
+        Removes the Access Control on a path and sub-paths.
+
+        :param acl:
+            Removes POSIX access control rights on files and directories.
+            The value is a comma-separated list of access control entries. Each
+            access control entry (ACE) consists of a scope, a type, and a user or
+            group identifier in the format "[scope:][type]:[id]”.
+        :type acl: str
+        :param progress_callback:
+            Callback where the caller can track progress of the operation
+            as well as collect paths that failed to change Access Control.
+        :type progress_callback: func(~azure.storage.filedatalake._generated.models.SetAccessControlRecursiveResponse)
+        :keyword str continuation:
+            Optional continuation token that can be used to resume previously stopped operation.
+        :keyword int batch_size:
+            Optional. If data set size exceeds batch size then operation will be split into multiple
+            requests so that progress can be tracked. Batch size should be between 1 and 2000.
+            The default when unspecified is 2000.
+        :keyword int timeout:
+            The timeout parameter is expressed in seconds.
+        :return: A summary of the recursive operations, including the count of successes and failures,
+            as well as a continuation token in case the operation was terminated prematurely.
+        :rtype: :class:`~azure.storage.filedatalake.models.AccessControlChanges`
+        """
+        if not acl:
+            raise ValueError("The Access Control List must be set for this operation")
+
+        options = self._set_access_control_recursive_options('remove', acl=acl, **kwargs)
+        return self._set_access_control_internal(options, progress_callback)
+
+    def _set_access_control_internal(self, options, progress_callback):
+        try:
+            total_directories_successful = 0
+            total_files_success = 0
+            total_failure_count = 0
+            continue_operation = True
+            while continue_operation:
+                headers, resp = self._client.path.set_access_control_recursive(**options)
+
+                if progress_callback is not None:
+                    progress_callback(resp)
```

Comment:
In .Net model of that callback changed recently - it now includes both batch counters and cumulative counts from start.

Reply:
Is it a requirement? I didn't add the cumulative counts as I felt like we are returning too many things for the cx to keep track of. I could add it if it's needed.

Reply:
It is not. That came out as a nice (and simple) addition during the review process.
For CX I'd argue that for long running ACL operations it relieves customer from doing math (depends how they want to report progress).